openmc/trackfile.py

```diff
@@ -0,0 +1,183 @@
+from collections import namedtuple
+
+import h5py
+
+from .checkvalue import check_filetype_version
+from .source import SourceParticle, ParticleType
+
+
+ParticleTrack = namedtuple('ParticleTrack', ['particle', 'states'])
+ParticleTrack.__doc__ = """\
+Particle track information
+
+Parameters
+----------
+particle : openmc.ParticleType
+    Type of the particle
+states : numpy.ndarray
+    Structured array containing each state of the particle. The structured array
+    contains the following fields: ``r`` (position; each direction in [cm]),
+    ``u`` (direction), ``E`` (energy in [eV]), ``time`` (time in [s]), ``wgt``
+    (weight), ``cell_id`` (cell ID) , and ``material_id`` (material ID).
+
+"""
+
+_VERSION_TRACK = 3
+
+
+def _identifier(dset_name):
+    """Return (batch, gen, particle) tuple given dataset name"""
+    _, batch, gen, particle = dset_name.split('_')
+    return (int(batch), int(gen), int(particle))
+
+
+class Track:
+    """Tracks resulting from a single source particle
+
+    Parameters
+    ----------
+    dset : h5py.Dataset
+        Dataset to read track data from
+
+    Attributes
+    ----------
+    identifier : tuple
+        Tuple of (batch, generation, particle number)
+    particles : list
```

Comment:
Just using this and I'm not sure ```particles``` is the right name to go for here.

We have ```settings.particles``` elsewhere in the code base and this ```tracks.particles``` appears to contain state information. I can't think of anything better but just wondering what others thought.

Reply:
I also struggled with nomenclature. If others have better ideas, I'm all ears!

Reply:
`Tracks.
particle_tracks` perhaps?

Reply:
I adopted @pshriwise's suggestion of `particle_tracks` (which is consistent with the `ParticleTrack` name used for objects within that list). I also ended up renaming the `TrackFile` class to `Tracks`.